packages/utils/src/abi_encoder/evm_data_types/method.ts

```diff
@@ -44,7 +44,19 @@ export class MethodDataType extends AbstractSetDataType {
         return returnValues;
     }
 
-    public getSignature(): string {
+    public decodeReturnValuesAsArray(returnData: string, rules?: DecodingRules): any[] {
```

Comment:
For all of these functions with an `any` return type (or variant), should the signatures include a generic type? E.g:

```
public decodeReturnValuesAsArray<T>(returnData: string, rules?: DecodingRules): T[]
```

Thoughts @fabioberger @LogvinovLeon @hysz ?